diff -rub ./libucontext-libucontext-1.2/Makefile ./libucontext-libucontext-1.2/Makefile
--- ./libucontext-libucontext-1.2/Makefile	2022-02-13 09:17:12.000000000 +0000
+++ ./libucontext-libucontext-1.2/Makefile	2024-06-09 12:26:41.133446980 +0100
@@ -17,7 +17,7 @@
 	override ARCH = aarch64
 endif
 
-LIBDIR := /lib
+LIBDIR := /usr/lib
 INCLUDEDIR := /usr/include
 PKGCONFIGDIR := /usr/lib/pkgconfig
 CFLAGS ?= -ggdb3 -O2 -Wall
@@ -198,10 +198,10 @@
 		install -D -m644 $$i ${DESTDIR}${INCLUDEDIR}/$$destfn; \
 	done
 	install -D -m644 ${LIBUCONTEXT_PC} ${DESTDIR}${PKGCONFIGDIR}/${LIBUCONTEXT_PC}
-	if [ -n ${LIBUCONTEXT_POSIX_NAME} ]; then \
-		install -D -m755 ${LIBUCONTEXT_POSIX_NAME} ${DESTDIR}${LIBUCONTEXT_POSIX_PATH}; \
-		install -D -m644 ${LIBUCONTEXT_POSIX_STATIC_NAME} ${DESTDIR}${LIBUCONTEXT_POSIX_STATIC_PATH}; \
-	fi
+	#if [ -n ${LIBUCONTEXT_POSIX_NAME} ]; then \
+	#	install -D -m755 ${LIBUCONTEXT_POSIX_NAME} ${DESTDIR}${LIBUCONTEXT_POSIX_PATH}; \
+	#	install -D -m644 ${LIBUCONTEXT_POSIX_STATIC_NAME} ${DESTDIR}${LIBUCONTEXT_POSIX_STATIC_PATH}; \
+	#fi
 
 install_docs: docs
 	install -D -m644 doc/libucontext.3 ${DESTDIR}/usr/share/man/man3/libucontext.3
