diff -rub ./SDL2_image-2.8.2/configure ./SDL2_image-2.8.2/configure
--- ./SDL2_image-2.8.2/configure	2024-01-01 21:35:15.000000000 +0000
+++ ./SDL2_image-2.8.2/configure	2024-06-09 22:39:56.300145586 +0100
@@ -5108,7 +5108,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | laylaos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -9896,6 +9896,17 @@
   hardcode_into_libs=yes
   ;;
 
+laylaos*)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
+  soname_spec='${libname}${release}${shared_ext}$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  dynamic_linker='LaylaOS ld.so'
+  hardcode_into_libs=yes
+  ;;
+
 # No shared lib support for Linux oldld, aout, or coff.
 linux*oldld* | linux*aout* | linux*coff*)
   dynamic_linker=no
