diff -rub ./libvorbis-1.3.7/configure ./libvorbis-1.3.7/configure
--- ./libvorbis-1.3.7/configure	2020-07-04 07:20:13.000000000 +0100
+++ ./libvorbis-1.3.7/configure	2024-08-26 18:17:18.828947341 +0100
@@ -5772,7 +5772,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | laylaos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -11392,6 +11392,17 @@
   hardcode_into_libs=yes
   ;;
 
+laylaos*)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
+  soname_spec='${libname}${release}${shared_ext}$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  dynamic_linker='LaylaOS ld.so'
+  hardcode_into_libs=yes
+  ;;
+
 # No shared lib support for Linux oldld, aout, or coff.
 linux*oldld* | linux*aout* | linux*coff*)
   dynamic_linker=no
