diff -rub ./ninja-1.12.1/src/util.cc ./ninja-1.12.1/src/util.cc
--- ./ninja-1.12.1/src/util.cc	2024-05-11 12:43:36.000000000 +0100
+++ ./ninja-1.12.1/src/util.cc	2024-12-14 09:22:26.319247366 +0000
@@ -802,7 +802,7 @@
     &mask) == 0) {
     return CPU_COUNT(&mask);
   }
-#elif defined(CPU_COUNT)
+#elif defined(CPU_COUNT) && !defined(__laylaos__)
   cpu_set_t set;
   if (sched_getaffinity(getpid(), sizeof(set), &set) == 0) {
     schedCount = CPU_COUNT(&set);
