diff -rub ./SDL2_net-2.2.0/configure ./SDL2_net-2.2.0/configure
--- ./SDL2_net-2.2.0/configure	2022-08-19 18:00:06.000000000 +0100
+++ ./SDL2_net-2.2.0/configure	2024-08-26 15:10:11.983867493 +0100
@@ -4930,7 +4930,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | laylaos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -9718,6 +9718,18 @@
   hardcode_into_libs=yes
   ;;
 
+# LaylaOS
+laylaos*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
+  soname_spec='${libname}${release}${shared_ext}$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  hardcode_into_libs=yes
+  ;;
+
 # No shared lib support for Linux oldld, aout, or coff.
 linux*oldld* | linux*aout* | linux*coff*)
   dynamic_linker=no
