diff -rub ./libtool-2.5.3/m4/libtool.m4 ./libtool-2.5.3/m4/libtool.m4
--- ./libtool-2.5.3/m4/libtool.m4	2024-09-25 14:52:26.000000000 +0100
+++ ./libtool-2.5.3/m4/libtool.m4	2024-11-19 18:39:38.264204327 +0000
@@ -2873,6 +2873,19 @@
   hardcode_into_libs=yes
   ;;
 
+# LaylaOS
+laylaos*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  dynamic_linker='LaylaOS ld.so'
+  hardcode_into_libs=yes
+  ;;
+
 # No shared lib support for Linux oldld, aout, or coff.
 linux*oldld* | linux*aout* | linux*coff*)
   dynamic_linker=no
@@ -3603,6 +3616,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+laylaos*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 # This must be glibc/ELF.
 linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
   lt_cv_deplibs_check_method=pass_all
@@ -5198,6 +5215,12 @@
       _LT_TAGVAR(link_all_deplibs, $1)=yes
       ;;
 
+    laylaos*)
+      _LT_TAGVAR(archive_cmds, $1)='$CC -shared $libobjs $deplibs $compiler_flags $wl-soname $wl$soname -o $lib'
+      _LT_TAGVAR(link_all_deplibs, $1)=yes
+      _LT_TAGVAR(ld_shlibs, $1)=yes
+      ;;
+
     os2*)
       _LT_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
       _LT_TAGVAR(hardcode_minus_L, $1)=yes
@@ -6997,6 +7020,12 @@
         _LT_TAGVAR(inherit_rpath, $1)=yes
         ;;
 
+      laylaos*)
+        _LT_TAGVAR(archive_cmds, $1)='$CC -shared $libobjs $deplibs $compiler_flags $wl-soname $wl$soname -o $lib'
+        _LT_TAGVAR(link_all_deplibs, $1)=yes
+        _LT_TAGVAR(ld_shlibs, $1)=yes
+        ;;
+
       linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
         case $cc_basename in
           KCC*)
